### PR TITLE
fix: thorchainSwapper check for EIP1559 support

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR ensures we check for EIP1559 support in THORChain swapper, effectively fixing Ledger THOR swaps off ETH.

tl;dr here is we checked for EIP1559 support at *quote* level, but didn't actually check for said support at execution level. The current version of `hw-app-eth` doesn't properly support EIP1559 fees, and we indeed have `ethSupportsEIP1559` have set to false in hdwallet-ledger.
As a result of the wrong fee scheme being used, Txs are broadcasted, but never picked in the mempool, which is why these were not visible in Etherscan.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes N/A

## Risk

Low - this should only fix Ledger, but a sanity check of any other wallet to ensure THOR swaps off ETH are still happy

## Testing

- Ledger THOR swaps off ETH are happy

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5613** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5613?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5613?utm_source=stack-comment).

File: src/lib/swapper/tradeExecution.ts [hunk 1]
```diff
@@ -32,12 +32,15 @@ export class TradeExecution {
...
       slippageTolerancePercentageDecimal,
-    }: CommonTradeExecutionInput,
+      ...rest
+    }: CommonTradeExecutionInput | EvmTransactionExecutionInput,
     buildSignBroadcast: (
       swapper: Swapper & SwapperApi,
       args: CommonGetUnsignedTransactionArgs,
     ) => Promise<string>,
   ) {
+    const supportsEIP1559 = 'supportsEIP1559' in rest ? rest.supportsEIP1559 : undefined
```
Comment (woodenfurniture) [inline src/lib/swapper/tradeExecution.ts:42]: not a fan of `in` for low-level performance reasons (irrelevant here), but also this (is relevant):
image

Comment (gomesalexandre) [reply src/lib/swapper/tradeExecution.ts:42]: I believe this one is fine, since `supportsEIP1559` is a boolean, it being undefined would evaluate to falsy, which is intended @woodenfurniture

**Review by kaladinlight — APPROVED** [review]
